File: 0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
       IList<IList<int>> result = new List<IList<int>>(); 
       if(numRows<1) return result;
       //Add 1 at start
       List<int> current = new List<int>();
       current.Add(1);
       result.Add(current);
       //2 - > numRows
       for(int i=2 ; i<=numRows;i++){
         int prev = 0;
         List<int> tempList = new List<int>();
         foreach(var num in current){
            int newVal = prev+num;
            tempList.Add(newVal);
            prev=num;
         }
         tempList.Add(1);//Add 1 at last
         result.Add(tempList);
         current=tempList;
       }
       return result;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        if numRows < 1:
            return result
        
        #Add 1 at start
        current = [1]
        result.append(current)

        for i in range(2,numRows + 1):
            prev = 0
            temp_list = []
            for num in current:
                new_val = prev + num
                temp_list.append(new_val)
                prev = num 
            temp_list.append(1) #Add 1 at last
            result.append(temp_list)
            current = temp_list
            
        return result